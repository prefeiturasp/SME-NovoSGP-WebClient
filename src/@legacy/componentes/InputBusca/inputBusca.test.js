import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import InputBusca from './InputBusca';

describe('InputBusca Component', () => {
  const mockOnClick = jest.fn();
  const mockOnPressEnter = jest.fn();
  const mockOnChange = jest.fn();

  const setup = (props = {}) => {
    const defaultProps = {
      label: 'Search Label',
      placeholder: 'Type something...',
      valor: '',
      onClick: mockOnClick,
      onPressEnter: mockOnPressEnter,
      onChange: mockOnChange,
      disabled: false,
      ...props,
    };
    return render(<InputBusca {...defaultProps} />);
  };

  it('should render the component with the label and input', () => {
    setup();
    expect(screen.getByText('Search Label')).toBeInTheDocument();
    expect(
      screen.getByPlaceholderText('Type something...')
    ).toBeInTheDocument();
  });

  it('should render the button inside the input prefix', () => {
    setup();
    expect(screen.getByRole('button')).toBeInTheDocument();
    expect(
      screen.getByRole('button').querySelector('i.fa-search')
    ).toBeInTheDocument();
  });

  it('should call onClick when the button is clicked', () => {
    setup();
    const button = screen.getByRole('button');
    fireEvent.click(button);
    expect(mockOnClick).toHaveBeenCalledTimes(1);
  });

  it('should call onChange when the input value changes', () => {
    setup();
    const input = screen.getByPlaceholderText('Type something...');
    fireEvent.change(input, { target: { value: 'new value' } });
    expect(mockOnChange).toHaveBeenCalledTimes(1);
    expect(mockOnChange).toHaveBeenCalledWith(expect.any(Object));
  });

  it('should disable the input when the disabled prop is true', () => {
    setup({ disabled: true });
    const input = screen.getByPlaceholderText('Type something...');
    expect(input).toBeDisabled();
  });

  xit('should match the snapshot', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });
});
