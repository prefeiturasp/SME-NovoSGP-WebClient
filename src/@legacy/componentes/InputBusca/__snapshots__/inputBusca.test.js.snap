// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputBusca Component should match the snapshot 1`] = `
<div>
  <div
    class="sc-fYdXmn ifMatY"
  >
    <div
      class="sc-beqWaB gnxwb"
    >
      <label
        class=""
        id="Search Label"
      >
        Search Label
      </label>
    </div>
    <span
      class="ant-input-affix-wrapper css-dev-only-do-not-override-1vtf12y"
    >
      <span
        class="ant-input-prefix"
      >
        <button
          class="ant-btn css-dev-only-do-not-override-1vtf12y ant-btn-link"
          type="button"
        >
          <i
            class="fa fa-search fa-lg"
          />
        </button>
      </span>
      <input
        class="ant-input css-dev-only-do-not-override-1vtf12y"
        placeholder="Type something..."
        type="text"
        value=""
      />
      <span
        class="ant-input-suffix"
      >
        <span
          class="ant-input-clear-icon ant-input-clear-icon-hidden"
          role="button"
          tabindex="-1"
        >
          <span
            aria-label="close-circle"
            class="anticon anticon-close-circle"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="close-circle"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
              />
            </svg>
          </span>
        </span>
      </span>
    </span>
  </div>
</div>
`;
